src/utils/env.ts

```diff
@@ -65,3 +76,20 @@ export const lookupPcEnv = (pcEnvTld: string, lenient = false, envTlds: string[]
 
     return null;
 };
+
+/**
+ * Attempts to locate a PC environment corresponding to the provided url
+ * @param url A string representing the Genesys Cloud environment top-level domain to search for
+ * @returns A Genesys Cloud environment object if found; null otherwise.
+ */
+export const lookupGcEnv = (url: string, targetEnv: string, envList: Environment[] = GC_DEV_EXTRA_ENVS): PcEnv|null => {
+    const hostLocation = new URL(url);
+    if (['localhost', '127.0.0.1'].indexOf(hostLocation.hostname) !== -1) {
+        return {
+            pcEnvTld: 'localhost',
+            pcAppOrigin: url
```

Comment:
maybe use `url.origin` here?  that way, we're assured to only use the origin in case extra path/query/hash data is accidentally included on the url